public/jira/derived/work-status/work-status.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module is repsonsible for determining the correct workType ("design", "dev", "qa", "uat")
+ * and statusType ("qa", "uat", "todo", "done", "blocked") for an issue.
+ */
+export const workType = ["design", "dev", "qa", "uat"] as const;
+export const workTypes = workType;
+
+export const workflowHappyPath = ["todo", "design", "dev", "qa", "uat", "done"];
+export const workflowUnhappyStatuses = ["blocked"];
+
+const inQAStatus = { qa: true, "in qa": true, "qa complete": true };
+const inPartnerReviewStatus = { "partner review": true, uat: true };
+const inDoneStatus = { done: true, cancelled: true };
+const blockedStatus = { blocked: true, delayed: true };
+
+export const inPartnerReviewStatuses = Object.keys(inPartnerReviewStatus);
+export const inIdeaStatus = { idea: true, "to do": true, open: true };
+export const inIdeaStatuses = Object.keys(inIdeaStatus);
+
+type WorkType = (typeof workType)[number];
+
+type Status =
+  | keyof typeof inQAStatus
+  | keyof typeof inPartnerReviewStatus
+  | keyof typeof inIdeaStatus
+  | keyof typeof inDoneStatus
+  | keyof typeof blockedStatus
+  | (string & {});
+
+type StatusCategory = "qa" | "uat" | "todo" | "done" | "blocked";
+
+export const statusCategoryMap = (function () {
+  const items = [
+    ["qa", inQAStatus],
+    ["uat", inPartnerReviewStatus],
+    ["todo", inIdeaStatus],
+    ["done", inDoneStatus],
+    ["blocked", blockedStatus],
+  ] as const;
+
+  const statusCategoryMap = {} as Record<Status, StatusCategory>;
+
+  for (let [category, statusMap] of items) {
+    for (let prop in statusMap) {
+      const status = prop;
+      statusCategoryMap[status] = category;
+    }
+  }
+
+  return statusCategoryMap;
+})();
+
+export function getStatusTypeDefault(issue: { status?: string }): StatusCategory | "dev" {
+  const statusCategory = statusCategoryMap[toLowerCase(issue?.status || "")];
+  if (statusCategory) {
+    return statusCategory;
+  } else {
+    return "dev";
+  }
+}
+
+function toLowerCase(str: string): string {
+  return str.toLowerCase();
+}
+
+const workPrefix = workType.map((wt) => wt + ":");
+
+function getWorkTypeDefault(normalizedIssue: { summary?: string; labels?: string[] }): WorkType {
+  let wp = workPrefix.find((wp) => (normalizedIssue?.summary || "").toLowerCase().indexOf(wp) === 0);
+
+  if (wp) {
+    return wp.slice(0, -1) as WorkType;
```

Comment:
I usually put a `nitpick` prefixes for comments that are more of a personal preference or feedback does not need to be addressed. So feel free to ignore any of the nitpick comments

nitpick: Probably an overkill in this scenario, especially that we are almost certain that the return values are of type `WorkType`. But do you think using a type guard serves any additional value?
Your solution is perfectly fine either way.
```
function isWorkType(value: string): value is WorkType {
  return workType.includes(value as WorkType);
}
```

Reply:
Ha, yeah, I went back and forth a lot here (and for the rest of the file). Ultimately I decided I wasn't going to update logic so I opted just to cast. Originally I was going to do something like:

```export const workType = ["design", "dev", "qa", "uat"] as const;
type WorkType = typeof workType[number]

const prefix = workType.map((wt): `${typeof wt}:` => `${wt}:`)

const slicePrefix = (prefixed: `${WorkType}:`): WorkType => {
  return prefixed.slice(0, -1) as WorkType
}


function getWorkTypeDefault(normalizedIssue: { summary?: string; labels?: string[] }): WorkType {
  const wp = prefix.find((wp) => (normalizedIssue?.summary || "").toLowerCase().indexOf(wp) === 0);

  if (wp) {
    return slicePrefix(wp)
  }

  const other = workType.find((wt) => normalizedIssue.labels?.map(str => str.toLocaleLowerCase()).includes(wt));

  if (other) {
    return other 
  }

  return "dev";
}ts

```
which is a nicer case than the current one. But because we don't change up the array I figured a cast wasn't a big deal here... Looking at some of the rest of the comments maybe I'll add it in. What are you thoughts. here it is in a [ts playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwgOwM7wO7QNYBUCeYwcAvHANoBEAJsKgJYDmylANHDcAG5scCOAQ16UArgJiUAunAGpEKdAG4AUDELEA6jgJFScNUQgAzOFih515ZCIC2AI2BRJy5UjTwwUYEfog9ZiyIAOhsBMAAKcIwYAEoALjgAAwASAG8DYGNTGABfOMTSAD4ktOi8xJiXN3Q4VAAbegRgAAUvHz8ycM9vX2BqBJTUrXMdYHL4uGHA4hJi1OU4OC8YEShkOG72vqD6xuBwgAZ2AFoARhiZOSnR5RyXZSMRZAQYehQ4RmAYa-UAEW8BCI6jBwshoKEGgAvPoASVQqBEwASqVqtlCUHwAH4EugoPRkIxFHA6gIHHVUNjajA8QTyNIchMfrp5otqpgwHpNr4gj5kNRIhgwBdZnBQeCBFDYfDEZidmiBBi4AAfJUcSgxIIwCAAGQgGEcAGFZPsNfjaCAAPJGKJC0gkMgHGIqBZwegmG0XFmLJZfVbrXZNVo9EAel13F1suAQGAAC0c-m06h5ZoFsSKcDBUAh9Gh1DhCOAQRJZIpITC4Vx6dxmp1EAQEuAuv1UCNqBNpuedREtFQUViTpcizdoujcagnpdi2WfqjsfjYcHPpWaw4tB4KhyQA)

Reply:
Its definitely not a big deal. But I like the new approach too. 
I did not know something like that is possible  `prefixed: `${WorkType}:`` 😮 

Should we move 
```
const prefix = workType.map((wt): `${typeof wt}:` => `${wt}:`)

const slicePrefix = (prefixed: `${WorkType}:`): WorkType => {
  return prefixed.slice(0, -1) as WorkType
}
```
Inside the function since thats where they are only being used?

Also thanks for addressing all of my comments!